static/js/filter.js

```diff
@@ -82,7 +82,12 @@ function message_matches_search_term(message, operator, operand) {
         if (page_params.realm_is_zephyr_mirror_realm) {
             return zephyr_stream_name_match(message, operand);
         }
-        return (message.stream.toLowerCase() === operand);
+        var stream_id = stream_data.get_stream_id(operand);
+        if (!stream_id) {
+            return false;
+        }
+
+        return (message.stream_id === stream_id);
```

Comment:
@showell this looks good, but I see one potential problem.

If you have an old message that was sent to a stream (later "deleted") in your home view history, I'm concerned that this will cause the narrow-to-stream button in the header of such a message to not work.

We should test that case and make sure that something sane happens.  Might find existing bugs with stream deletion that we'll want to open; I just want to make sure this doesn't regress badly.